Source/WebCore/dom/SlotAssignment.cpp

```diff
@@ -387,6 +388,10 @@ void NamedSlotAssignment::willRemoveAssignedNode(const Node& node, ShadowRoot&)
     slot->assignedNodes.removeFirstMatching([&node](const auto& item) {
         return item.get() == &node;
     });
+
+    std::optional<Style::PseudoClassChangeInvalidation> styleInvalidation;
+    if (auto* slotElement = slot->element.get())
+        styleInvalidation.emplace(*slotElement, CSSSelector::PseudoClass::HasSlotted, slotElement->hasFlattenedSlottedContent());
```

Comment:
Is there a way to share more of this code? This sequence of three lines of code is pretty repetitive, shows up 5 times almost identically, and I think a helper function could improve clarity.

Reply:
`Style::PseudoClassChangeInvalidation` is scope-dependent. It would be possible to define another scope that wraps this scope, but not worth it IMO